testing/test_unraisableexception.py

```diff
@@ -236,6 +251,68 @@ def test_it():
     result.stderr.fnmatch_lines("ValueError: del is broken")
 
 
+@pytest.mark.filterwarnings("default::pytest.PytestUnraisableExceptionWarning")
+def test_refcycle_unraisable_warning_filter(pytester: Pytester) -> None:
+    # note that the host pytest warning filter is disabled and the pytester
+    # warning filter applies during config teardown of unraisablehook.
+    # see: https://github.com/pytest-dev/pytest/issues/10404
+    pytester.makepyfile(
+        test_it="""
+        import pytest
+
+        class BrokenDel:
+            def __init__(self):
+                self.self = self  # make a reference cycle
+
+            def __del__(self):
+                raise ValueError("del is broken")
+
+        def test_it():
+            BrokenDel()
+        """
+    )
+
+    with _disable_gc():
+        result = pytester.runpytest("-Werror")
+
+    # TODO: should be a test failure or error
+    assert result.ret == pytest.ExitCode.INTERNAL_ERROR
+
+    result.assert_outcomes(passed=1)
+    result.stderr.fnmatch_lines("ValueError: del is broken")
+
+
+@pytest.mark.filterwarnings("default::pytest.PytestUnraisableExceptionWarning")
+def test_create_task_raises_unraisable_warning_filter(pytester: Pytester) -> None:
+    # note that the host pytest warning filter is disabled and the pytester
+    # warning filter applies during config teardown of unraisablehook.
+    # see: https://github.com/pytest-dev/pytest/issues/10404
+    # This is a dupe of the above test, but using the exact reproducer from
+    # the issue
+    pytester.makepyfile(
+        test_it="""
+        import asyncio
+        import pytest
+
+        async def my_task():
+            pass
+
+        def test_scheduler_must_be_created_within_running_loop() -> None:
+            with pytest.raises(RuntimeError) as _:
+                asyncio.create_task(my_task())
+        """
+    )
+
+    with _disable_gc():
+        result = pytester.runpytest("-Werror")
+
+    # TODO: should be a test failure or error
+    assert result.ret == pytest.ExitCode.INTERNAL_ERROR
+
+    result.assert_outcomes(passed=1)
+    result.stderr.fnmatch_lines("RuntimeWarning: coroutine 'my_task' was never awaited")
```

Comment:
If we run this without `-Werror` we can check that normal warning functionality works.

Reply:
ah I broke it! Very good catch! I'll send a commit to fix it now